src/go/util/url_util_test.go

```diff
@@ -446,14 +446,34 @@ func TestWildcardMatcherForPath(t *testing.T) {
 			wantMatcher: "",
 		},
 		{
-			desc:        "Multiple path params",
+			desc:        "Path params with fieldpath-only bindings",
 			uri:         "/shelves/{shelf_id}/books/{book.id}",
-			wantMatcher: `/shelves/[^\/]+/books/[^\/]+$`,
+			wantMatcher: `^/shelves/[^\/]+/books/[^\/]+$`,
 		},
 		{
-			desc:        "Path param with wildcard",
+			desc:        "Path params with full segment binding",
+			uri:         "/v1/{name=books/*}",
+			wantMatcher: `^/v1/[^\/]+$`,
+		},
+		{
+			desc:        "Path param with wildcard segments",
+			uri:         "/test/*/test/**",
+			wantMatcher: `^/test/[^\/]+/test/.*$`,
+		},
+		{
+			desc:        "Path param with wildcard in segment binding",
 			uri:         "/test/{x=*}/test/{y=**}",
-			wantMatcher: `/test/[^\/]+/test/[^\/]+$`,
+			wantMatcher: `^/test/[^\/]+/test/.*$`,
+		},
+		{
+			desc:        "Path param with mixed wildcards",
+			uri:         "/test/{name=*}/test/**",
+			wantMatcher: `^/test/[^\/]+/test/.*$`,
```

Comment:
can we support following two cases?
/foo/*/bar
/foo/**/bar

Reply:
- /foo/*/bar is supported, it's tested on line 460.
- /foo/** is supported, it's tested on line 460 as well.
- /foo/**/bar is also technically supported, but it is actually not valid syntax. From the HttpRule documentation:

> The syntax * matches a single URL path segment. The syntax ** matches zero or more URL path segments, which must be the last part of the URL path except the Verb.

But it turns out that our path matcher does support it:

https://github.com/GoogleCloudPlatform/esp-v2/blob/af02e4ecce9401f3287360ea12d52dcfab579ee3/src/api_proxy/path_matcher/http_template_test.cc#L218

So what would you recommend here? Explicitly support `**` where it's not the last path segment? Or just treat is as not supported?

Reply:
Well,  if htttp rule document say not need to support /foo/**/bar, then we don't need to support it.